core/shared/src/test/scala/fs2/StreamCombinatorsSuite.scala

```diff
@@ -1003,6 +1003,17 @@ class StreamCombinatorsSuite extends Fs2Suite {
       .map(results => assert(results.size == 3))
   }
 
+  test("meteredStartImmediately should not wait between events that last longer than the rate") {
+    Stream
+      .eval[IO, Int](IO.sleep(1.second).as(1))
+      .repeatN(10)
+      .meteredStartImmediately(1.second)
+      .interruptAfter(4500.milliseconds)
+      .compile
+      .toList
+      .map(results => assert(results.size == 4))
+  }
```

Comment:
Not specific to your PR, but seems like we should make more use of `TestControl.executeEmbed(...)` in this suite for time-based tests.